platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/event/EventDescriptor.java

```diff
@@ -48,96 +47,103 @@ private static final class ClassVersion {
     private NodeId creator;
     private long generation;
 
-    private int hashCode;
-
-    public ChatterEventDescriptor() {}
+    /**
+     * Zero arg constructor, required for deserialization. Do not use manually.
+     */
+    public EventDescriptor() {}
 
     /**
-     * Create a new gossip event descriptor.
+     * Create a new event descriptor.
      *
      * @param hash       the hash of the event
      * @param creator    the creator of the event
      * @param generation the age of an event, smaller is older
      */
-    public ChatterEventDescriptor(@NonNull final Hash hash, @NonNull final NodeId creator, final long generation) {
+    public EventDescriptor(@NonNull final Hash hash, @NonNull final NodeId creator, final long generation) {
         this.hash = Objects.requireNonNull(hash, "hash must not be null");
         this.creator = Objects.requireNonNull(creator, "creator must not be null");
         this.generation = generation;
-
-        hashCode = Objects.hash(hash, creator, generation);
```

Comment:
The purpose of this was to make the code simpler. I think it's a lot simpler just to use `Hash.hashCode()` as the hash of a descriptor. Event hashes are presumably unique. And even if two events somehow have the same hash, the contract for `int hashCode()` allows for unequal objects to have the same hash codes. Having hash collisions like this can hurt performance of various data structures if they happen too frequently. But since it's infinitesimally unlikely here, I think the risk is negligible.

Once I made this simplification, Lazar suggested I didn't bother caching the hash code, since it's really quite cheap to just ask the `Hash` object whenever we end up needing it.